pkg/util/mount/mount_linux.go

```diff
@@ -160,10 +160,12 @@ func (*Mounter) List() ([]MountPoint, error) {
 
 // IsLikelyNotMountPoint determines if a directory is not a mountpoint.
 // It is fast but not necessarily ALWAYS correct. If the path is in fact
-// a bind mount from one part of a mount to another it will not be detected.
+// a bind mount from one part of a mount to another it will not be detected, unless
+// the source itself is a mount point.
```

Comment:
Ack. Although the comment should involve something related to device names of the path and it's parent.

It's not immediately clear what the function does and it takes a while to understand why the bind mount check doesn't work and also why in some cases it does.